### DISCUSSION TITLE
Toolbar Inside a Toolbar (Groups Items or Item->Sub Item)

### DISCUSSION THREAD

**Discussion by FelipeRearden** [chrisgurney/obsidian-note-toolbar#38] [Ideas]
Hello @chrisgurney !!!!!!!

I have this idea that might be a good way to deal with Toolbars that has a lot of items.

In a nutshell: the idea is to be able to access a toolbar as an item from another toolbar.
ℹ️ Transform a Toolbar in a item;
ℹ️ This new `item-toolbar` will be deal as an item;
ℹ️ When I click is this `item-toolbar` it will open the the items from the toolbar

## Example 

Let use an example to explain this is detail.

📦 This is my database

1️⃣ Toolbar 1 `TB1`
🔵 Item `1A`
🔵 Item `1B`

2️⃣ Toolbar 2 `TB2`
🔴 Item `2A`
🔴 Item `2B`

3️⃣ Toolbar 3 `TB3`
🟡 Item `3A`
🟡 Item `3B`

✅ **I want to add a new ITEM to** `TB3` 
✅ **The ITEM that I want to Add is** all items from `TB2`

After adding `TB2` to `TB3` this will be my database:

1️⃣ Toolbar 1 `TB1`
🔵 Item `1A`
🔵 Item `1B`

2️⃣ Toolbar 2 `TB2`
🔴 Item `2A`
🔴 Item `2B`

3️⃣ Toolbar 3 `TB3`
🟡 Item `3A`
🟡 Item `3B`
2️⃣ Toolbar 2 `TB2`

🚨 When I open `TB3` in th active Note I will get this

```
🟡 3A 🟡 3B 2️⃣ TB2
```

🚨 When I CLICK/ENTER on item `TB2` it will open all items from the toolbar `TB2` **REPLACING the content from TB3**

```
🔴 2A 🔴 2B
```

Conclusion: this way we have grouped item inside the toolbar where the criteria to group items is the another toolbars !!!!

## Development Suggestion

While is was thinking about this FR I was trying to help you with an idea on how to implement this **without hurting the status quo** 

Would be amazing if you could implement this using the amazing structure that you already created an works beautifuly.

My idea 

On Edit Toolbar UI… 

1️⃣ A 4th type to `Command - File - URI` array called `Toolbar`

2️⃣ Text field will trigger all Toolbars from the database for the user to choose

3️⃣ User will choose the toolbar the will be triggerd

4️⃣ Since it is an item you can define an icon for the "toolbar"

In a nutshell: use the item structure that we already have to connect items from one toolbar to another toolbar

## Info

Maybe, maybe this idea could help to implement #31 

My idea is to merge toolbar where the second toolbar is a **single item**

The idea from #31 is merge where the second toolbar items is added to the first toolbar

---

Thanks for reading this !!!!! I hope you like my suggestion 🙏 🙏 🙏

Let me know if you need more information !!!

Have a great day ☀️

**Comment by MrComputron** [comment]
I would love the ability to nest Toolbars just for the sake of better management, especially if you can select where the toolbar line break is.

For example if I have a set of general editing commands I want to have visible in all files, in the current implementation I need to update all toolbars to include them.

If I can nest toolbar - or add them as line items to a toolbar - then I can create the toolbar for just these general editing items, and add it as a single toolbar item to every toolbar I want it to be present in.

If I can choose the vertical layout of rows for a toolbar then I can set it as the first item so it's always at the top of the toolbar regardless of subsequent toolbar.

**Comment by FelipeRearden** [comment]
From https://github.com/chrisgurney/obsidian-note-toolbar/discussions/34#discussioncomment-9346836

> As a step 1, I'll likely be adding menu support, as it works well on all screen sizes, and I've already done a lot of the foundational work (as it was needed for the floating buttons).

@chrisgurney based on your recents comments about a new `Menu` type, this is my humble attempt to add some light to this FR for you to evaluate if is a good fit for Note Toolbar 🙏

**Comment by chrisgurney** [comment]
Another vote for this enhancement over here: #77

**Reply by FelipeRearden** [reply]
Thanks for the info @chrisgurney !!!! 

I am glad to know more people on the community has the same  principle about multiple toolbars inside a toolbar 👍

🙏 🙏 🙏

**Reply by FeralFlora** [reply]
Oops, I should have searched more thoroughly, although I think my suggestion is more in line with #32, since the current discussion is about representing other toolbars as a single button, whereas what I wrote was about seamlessly merging toolbars, if one is set to be "additive".

**Comment by chrisgurney** [comment]
I'm going to close this issue -- but keep others open -- as this more or less spoke to embedding one toolbar within another with _Item Groups_. Clicking an item to expand is similar in concept to _Item Menus_.

**Comment by MrComputron** [comment]
Great work Chris. Note toolbar becomes more functional and polished every week!

**Comment by MrComputron** [comment]
You mad man, you did it! @chrisgurney 
![image](https://github.com/user-attachments/assets/767b376c-5370-40a0-b79b-5f455ef037b5)
![image](https://github.com/user-attachments/assets/88886c75-1886-49aa-a9b2-f8456aee686c)

The new item type "Item Group" is perfect! I can't believe how far you've pushed Note Toolbar in just a few months. Thanks for all the hard work you've put into this fantastic plugin!
